caproto/server/common.py

```diff
@@ -408,21 +408,30 @@ def get_db_entry():
                                       ca.DEFAULT_PROTOCOL_VERSION)
                 ]
         elif isinstance(command, ca.CreateChanRequest):
+            pvname = command.name
             try:
-                db_entry = self.context[command.name]
+                db_entry = self.context[pvname]
             except KeyError:
                 self.log.debug('Client requested invalid channel name: %s',
-                               command.name)
+                               pvname)
                 to_send = [ca.CreateChFailResponse(cid=command.cid)]
             else:
 
                 access = db_entry.check_access(self.client_hostname,
                                                self.client_username)
 
+                modifiers = ca.parse_record_field(pvname).modifiers
+                data_type = db_entry.data_type
+                data_count = db_entry.max_length
+                if ca.RecordModifiers.long_string in (modifiers or {}):
+                    if data_type in (ChannelType.STRING, ):
```

Comment:
Should this check whether the _native_ type is STRING and then, if so, find the corresponding CHAR type?

Reply:
I believe `db_entry` (`ChannelData`) `.data_type` will never be a non-native type.

Reply:
Ah, of course.